frontend/src/app/modules/challenges/challenges-test-page/challenges-test-page.component.ts

```diff
@@ -1,42 +1,142 @@
-import { ChangeDetectorRef, Component } from '@angular/core';
-import { DropdownItem } from '@shared/models/dropdown-item';
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { fillFormInputs } from '@modules/challenges/challenge-creation/challenge-creation.utils';
+import { NavigationTabType } from '@shared/enums/navigation-tab-type';
+import { NavigationTab } from '@shared/models/navigation-tab';
+
+import { tabs } from './challenges-test-page.utils';
 
 @Component({
     selector: 'app-challenges-test-page',
     templateUrl: './challenges-test-page.component.html',
     styleUrls: ['./challenges-test-page.component.sass'],
 })
-export class ChallengesTestPageComponent {
-    public languagesList: DropdownItem[] = [{ content: 'JavaScript', iconName: 'node-js' }, { content: 'C#' }];
+export class ChallengesTestPageComponent implements OnInit, OnChanges {
+    @Input() editorLanguage: string;
+
+    @Input() testCases: string;
+
+    @Input() exampleTestCases: string;
+
+    @Input() checkValidation = false;
 
-    public languageVersionsList: DropdownItem[] = [{ content: 'Node v18.x' }, { content: 'Node v17.x' }];
+    @Output() testCasesChange = new EventEmitter<string>();
 
-    public tabs = [
-        { title: 'Test Cases', active: true },
-        { title: 'Example Test Cases', content: 'Example Test Cases content' },
-        { title: 'Help', content: 'Help infomation' },
-    ];
+    @Output() exampleTestCasesChange = new EventEmitter<string>();
 
-    public selectedLanguage: string;
+    @Output() validationChange = new EventEmitter<boolean>();
+
+    public editorContent: string;
+
+    public tabs = tabs;
+
+    public currentTab = tabs[0];
 
     public isMaximized: boolean = false;
 
-    private languageNameMapping: { [key: string]: string } = {
-        'C#': 'csharp',
-    };
+    public inputForm = new FormGroup({
+        testCases: new FormControl('', [
+            Validators.required,
+        ]),
+        exampleTestCases: new FormControl('', [
+            Validators.required,
+        ]),
+    });
 
-    constructor(private cdr: ChangeDetectorRef) {}
+    ngOnInit(): void {
+        this.editorContent = this.testCases;
+    }
 
-    toggleMaximize() {
-        this.isMaximized = !this.isMaximized;
+    public onCodeChange(code: string) {
+        this.editorContent = code;
+
+        switch (this.currentTab.type) {
+            case NavigationTabType.TestCases:
+                this.testCases = code;
+                this.testCasesChange.emit(code);
+
+                this.inputForm.controls.testCases.setValue(code);
+                this.validationChange.emit(this.inputForm.valid);
```

Comment:
dublicates on 67 line, i guess it can be moved out of switch

Reply:
Fixed